### PR TITLE
chore: add GN linting

### PR DESCRIPTION
**Pull request by nornagon**
##### Description of Change
Runs `gn format` over `*.gn` and `*.gni` files. Use `--fix` to fix the errors automatically.

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: no-notes

### CODE REVIEW

**Comment by nornagon** [comment]
d'oh, the lint CI job doesn't run from a `gclient sync`'d directory, so it doesn't have a copy of `gn`. I'll see if I can fix that without having to get a full gclient sync. WIP.

**Comment by nornagon** [comment]
This is working now! The lint errors are legit errors. The bash stuff is a bit crazy though.

File: .vsts/lint.yml [hunk 1]
```diff
@@ -13,7 +13,17 @@ steps:
   name: Setup_depot_tools
 
 - bash: |
+    chromium_revision="$(grep -A1 chromium_version DEPS | tr -d '\n' | cut -d\' -f4)"
+    buildtools_revision="$(curl -sL "https://chromium.googlesource.com/chromium/src/+/${chromium_revision}/DEPS?format=TEXT" | base64 -d | grep buildtools_revision -A1 | tr -d '\n' | cut -d\' -f4)"
```
Comment (MarshallOfSound) [inline .vsts/lint.yml]: This is wild and I love it 😆

Comment (alexeykuzmin) [reply .vsts/lint.yml]: I will be broken in less than a month =)

**Pull request merged**